lib/msf/core/exploit/format/rar_symlink_path_traversal.rb

```diff
@@ -0,0 +1,75 @@
+# Encoding: ASCII-8BIT
+
+module Msf
+  class Exploit
+    module Format
+      # The RarSymlinkPathTraversal mixin provides methods for generating a RAR file
+      # that exploits CVE-2022-30333, which can write an arbitrary file to an arbitrary
+      # location on a Linux filesystem
+      module RarSymlinkPathTraversal
+        def initialize(info = {})
+          super
+        end
+
+        # Encode arbitrary data to be extracted to an arbitrary path on versions of
+        # unrar that are affected by CVE-2022-30333
+        def encode_as_traversal_rar(symlink_name, target_path, data)
+          unless target_path.length < 0x68
```

Comment:
```suggestion
          unless target_path.length < 103
```

Reply:
I think <104 is correct - we have 104 characters and need room for a null-terminator